azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -209,18 +209,31 @@ public FlowContainer(final Props props,
   public static void main(final String[] args) throws ExecutorManagerException {
     // Redirect all std out and err messages into slf4j
     StdOutErrRedirect.redirectOutAndErrToLog();
-    // Get all the arguments
-    final String execIdStr = args[EXEC_ID_INDEX];
+    // Get the execution ID from the environment
+    final String execIdStr = System.getenv(FLOW_EXECUTION_ID);
+    if (execIdStr == null) {
+      String msg = "Execution ID is not set!";
+      logger.error(msg);
+      throw new ExecutorManagerException(msg);
+    }
     // Process Execution ID.
     int execId = -1;
     try {
       execId = Integer.parseInt(execIdStr);
+      logger.info("Execution ID : " + execId);
     } catch (NumberFormatException ne) {
       logger.error("Execution ID passed in argument is invalid {}", execIdStr);
       throw new ExecutorManagerException(ne);
     }
 
-    logger.info("Execution ID : " + execId);
+    // Get the version set id.
+    final String versionSetId = System.getenv(VERSION_SET_ID);
+    if (versionSetId == null) {
+      // Should not happen
+      logger.warn("VersionSet ID is not set!");
+    } else {
+      logger.info("VersionSet ID : {}", versionSetId );
```

Comment:
Can you please print version set json over here? It will give detail about which image version is used for each type.

Reply:
+1

Reply:
Will do.

Reply:
Env variable is only id. You need to fetch version set json from database.